improver/utilities/save.py

```diff
@@ -107,6 +107,11 @@ def save_netcdf(cubelist, filename, compression_level=1, least_significant_digit
     for cube in cubelist:
         _order_cell_methods(cube)
         _check_metadata(cube)
+        # iris.fileformats.netcdf.save will add a new "least_significant_digit"
+        # attribute, but will not update an existing attribute when saving with
+        # different precision.  Therefore we remove the "least_significant_digit"
+        # attribute if present.
+        cube.attributes.pop("least_significant_digit", None)
```

Comment:
Can you not surround this with `if least_significant_digit is not None:`? As I understand it, as written this PR means that any improver operation that doesn't explicitly state --least-significant-digit will discard the least significant digit information.

Reply:
The beauty of a 2 review system is that you have to explain and agree everything twice...

Reply:
It would be better to remove this attribute.  The reason is that whenever we manipulate the data within improver (eg thresholding, neighbourhooding, lapse rate correction - essentially anything apart from the "standardise (metadata only)" and "extract" steps) this "precision" information ceases to be correct, but is not removed.  This might be better addressed through additions to `BasePlugin` in the longer term.  However it has the same effect to remove it here, and since this is the one place in improver with "knowledge" of NetCDF attributes specifically it seemed a sensible place to do this.

Reply:
Certainly when we threshold to 1K we are not a lot more accurate than 1K. However, are we really going to keep track of this throughout each processing step? I'd say we're not, and I think we should leave the attribute as is but add a change so that superseding is possible. Probably a bit more typical for a NetCDF tool to leave it unless instructed otherwise.

Reply:
Once again, I'm going to pass this to @bjwheltor .  In my view we shouldn't retain incorrect attributes, and I don't fully understand why thresholding a temperature field would mean the `least_significant_digit` on the input would remain correct.  Perhaps we should clarify offline.

Reply:
Email exchange has now confirmed @bjwheltor  and @benfitzpatrick  have both agreed the current implementation is the correct behaviour